bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -46,6 +51,14 @@
   private Context operationContext;
   private Context rpcContext;
 
+  private RetryOptions retryOptions;
+
+  // The number of times we've retried after a timeout
+  private AtomicInteger timeoutRetryCount = null;
```

Comment:
This was the way we did things, which I refactored for this PR, but didn't look at too carefully.  This code was updated over time by different people, and I didn't look at it too carefully.  I'm glad you're reviewing it.

I think synchronizing the handle exception methods is a better approach... an int would work fine in that case.